src/modules/hip/kernel/resize.hpp

```diff
@@ -0,0 +1,735 @@
+    d_float16 locSrc_f16;
+    resize_roi_and_srclocs_hip_compute(&srcRoi_i4, &dstDimsWH, id_x, id_y, &locSrc_f16);
+
+    d_float24 dst_f24;
+    rpp_hip_interpolate24_bilinear_pln3(srcPtr + srcIdx, &srcStridesNCH, &locSrc_f16, &srcRoi_i4, &dst_f24, false);
+    rpp_hip_pack_float24_pln3_and_store24_pkd3(dstPtr + dstIdx, &dst_f24);
+}
+
+template <typename T>
+__global__ void resize_generic_pkd_tensor(T *srcPtr,
+                                          uint2 srcStridesNH,
+                                          T *dstPtr,
+                                          uint2 dstStridesNH,
+                                          RpptImagePatchPtr dstImgSize,
+                                          RpptROIPtr roiTensorPtrSrc,
+                                          RpptInterpolationType interpolationType)
+{
+    int id_x = hipBlockIdx_x * hipBlockDim_x + hipThreadIdx_x;
+    int id_y = hipBlockIdx_y * hipBlockDim_y + hipThreadIdx_y;
+    int id_z = hipBlockIdx_z * hipBlockDim_z + hipThreadIdx_z;
+
+    uint2 dstDimsWH;
+    dstDimsWH.x = dstImgSize[id_z].width;
+    dstDimsWH.y = dstImgSize[id_z].height;
+
+    if ((id_y >= dstDimsWH.y) || (id_x >= dstDimsWH.x))
+    {
+        return;
+    }
+
+    int4 srcRoi_i4 = *(int4 *)&roiTensorPtrSrc[id_z];
+    uint2 srcDimsWH;
+    srcDimsWH.x = srcRoi_i4.z - srcRoi_i4.x + 1;
+    srcDimsWH.y = srcRoi_i4.w - srcRoi_i4.y + 1;
+
+    int widthLimit = srcRoi_i4.z * 3;
+    int heightLimit = srcRoi_i4.w;
+    float wRatio = (float)srcDimsWH.x / (float)dstDimsWH.x;
+    float hRatio = (float)srcDimsWH.y / (float)dstDimsWH.y;
+    float hScale = 1.0f, wScale = 1.0f, hRadius = 1.0f, wRadius = 1.0f;
+
+    rpp_hip_compute_interpolation_scale_and_radius(interpolationType, srcDimsWH.x, dstDimsWH.x, &wScale, &wRadius, wRatio);
+    rpp_hip_compute_interpolation_scale_and_radius(interpolationType, srcDimsWH.y, dstDimsWH.y, &hScale, &hRadius, hRatio);
+    float wOffset = (wRatio - 1) * 0.5f - wRadius;
+    float hOffset = (hRatio - 1) * 0.5f - hRadius;
+    int wKernelSize = ceilf(wRadius * 2);
+    int hKernelSize = ceilf(hRadius * 2);
+
+    float rowWeight, colWeight, rowCoeff, colCoeff;
+    int srcLocationRowFloor, srcLocationColumnFloor;
+    resize_roi_generic_srcloc_and_weight_hip_compute(srcRoi_i4.x, id_x, wRatio, widthLimit, &srcLocationColumnFloor, &colWeight, wOffset, 3);
+    resize_roi_generic_srcloc_and_weight_hip_compute(srcRoi_i4.y, id_y, hRatio, heightLimit, &srcLocationRowFloor, &rowWeight, hOffset, 1);
+
+    T *srcPtrTemp = srcPtr + (id_z * srcStridesNH.x);
+    float3 outPixel_f3 = (float3)0.0f;
+    float rowCoeffSum = 0.0f, colCoeffSum = 0.0f;
+    for(int j = 0; j < hKernelSize; j++)
+    {
+        int rowIndex = fminf(fmaxf((int)(srcLocationRowFloor + j), 0), heightLimit);
+        T *srcRowPtrsForInterp = srcPtrTemp + rowIndex * srcStridesNH.y;
+        rpp_hip_compute_interpolation_coefficient(interpolationType, (rowWeight - hRadius + j) * hScale , &rowCoeff);
```

Comment:
probably you can update the rowCoeffSum in the function itself instead of storing it in a separate variable